src/gc/boehm.cr

```diff
@@ -5,9 +5,17 @@
 # MUSL: On musl systems, libpthread is empty. The entire library is already included in libc.
 # The empty library is only available for POSIX compatibility. We don't need to link it.
 #
+# Darwin: `libpthread` is provided as part of `libsystem`. There's no reason to link it explicitly.
+#
+# Interpreter: Starting with glibc 2.34, `pthread` is integrated into `libc`
+# and may not even be available as a separate shared library.
+# There's always a static library for compiled mode, but `Crystal::Loader` does not support
+# static libraries. So we just skip `pthread` entirely. The symbols are still
+# available in the interpreter because they are loaded in the compiler.
+#
 # OTHERS: On other systems, we add the linker annotation here to make sure libpthread is loaded
 # before libgc which looks up symbols from libpthread.
-{% unless flag?(:win32) || flag?(:musl) %}
+{% unless flag?(:win32) || flag?(:musl) || flag?(:darwin) || (flag?(:interpreted) && flag?(:gnu)) %}
```

Comment:
Because this fix is specific to the interpreter linked against glibc. On other systems we probably need to link `pthread` explictly.

Note that this condition is also true when the interpreter runs on win32, musl or darwin.
So in terms of currently supported systems, this branch essentially only applies to BSDs.